teradata/datadog_checks/teradata/check.py

```diff
@@ -0,0 +1,176 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import json
+import time
+from contextlib import closing, contextmanager
+
+try:
+    import teradatasql
+
+    TERADATASQL_IMPORT_ERROR = None
+except ImportError as e:
+    teradatasql = None
+    TERADATASQL_IMPORT_ERROR = e
+
+
+from datadog_checks.base import AgentCheck, is_affirmative
+from datadog_checks.base.constants import ServiceCheck
+from datadog_checks.base.utils.db import QueryManager
+
+from .config_models import ConfigMixin
+from .queries import COLLECT_RES_USAGE, DEFAULT_QUERIES
+
+SERVICE_CHECK_CONNECT = 'can_connect'
+SERVICE_CHECK_QUERY = 'can_query'
+
+
+class TeradataCheck(AgentCheck, ConfigMixin):
+    __NAMESPACE__ = 'teradata'
+
+    def __init__(self, name, init_config, instances):
+        super(TeradataCheck, self).__init__(name, init_config, instances)
+
+        self._connect_params = None
+        self._connection = None
+
+        self._tags = [
+            'teradata_server:{}'.format(self.instance.get('server')),
+            'teradata_port:{}'.format(self.instance.get('port', 1025)),
+        ]
+        self._tags.extend(self.instance.get('tags', []))
```

Comment:
populate in `initialize_config`?

Reply:
If you do that then query manager is not going to have the right tags on line 53

Reply:
I was able to populate and update `query_manager.tags` in `initialize_config`: 
https://github.com/DataDog/integrations-core/pull/11701/files#diff-920339eef5a5eedd2f6f2139b7854a25a988b1cd3aae145ed856aff977b4e5e8R94-R96